### PR TITLE
Mergin all together

### LINKED ISSUE #94: Create elections page
**Is your feature request related to a problem? Please describe.**
We want to have a different page for showing the elections. This is the first part of it: creating the page with the current design.

**Describe the solution you'd like**
- A copy of the current ballots page
- The page only shows the elections (already removed from the ballots page)
- The filter doesn't allow filter by `type`
- There is an `Elections` link on the top menu

**Describe alternatives you've considered**
A simplified version of the filter should be displayed. This filter should be the same component, but a variant that doesn't show the type filter.

### LINKED ISSUE #105: Ballots page does not change DAO filter using 'treasury' URL parameter
From the DAOs page you can click on any **Opened Decisions** button and you should be redirected to ballots page with the `treasury` and `active` filters.

What's actually happening is that the `active` filter is always `on` by default but the `treasury` param is never applied. So you end up watching a list of other ballots, not what you expected. 

image

### LINKED ISSUE #114: Ballot creation fails due to lack of funds even though the user has enough.
**Describe the bug**   
The ballot creation form does not query properly the balance of the user and therefore always fails.

**To Reproduce**
1. Login with an account with more than 10.0000 TLOS
2. Go to the 'Proposals' Page
3. Click on the 'Create a Poll' Button (you may need to run an older version because right now the button was removed temporarily).
4. Fill all the fields and chose VOTE as treasury.
5. click Create
6. This dialog popups with the failure text.

**Expected behavior**
- It must work when the user actually has enough funds.
- If the user actually does not have enough funds, the text should say how much is needed and how much the user has, for a more friendly interaction.

**Screenshots**

image

### PR DESCRIPTION
**Pull request by Viterbo**
# Description
This branch has been rebased with [116-proposals-link-is-not-highlighted...](https://github.com/telosnetwork/app-telos-native/tree/116-proposals-link-is-not-highlighted-when-in-the-main-page) and [114-ballot-creation-fails...](https://github.com/telosnetwork/app-telos-native/tree/114-ballot-creation-fails-due-to-lack-of-funds-even-though-the-user-has-enough)
Therefore, this fixes all the issues fixed there.
Also adding some extra fixes.

Fixes #94, #120, #107, #119, #113 and #116 (taken from 116-proposals-link-is-not-highlighted-when-in-the-main-page)
Fixes #114 (taken from 114-ballot-creation-fails-due-to-lack-of-funds-even-though-the-user-has-enough)
Fixes #105, #112 (by itself)

## Changes
Other changes that were made but have no issue associated are:
- Ballots are never deleted one fetched, just not shown when using filters. This eliminates some unnecessary waits.
- In BallotView the `iframeUrl` function now prioritizes the PDF if present instead of the main image.

### CODE REVIEW

**Review by karynemayer — APPROVED** [review]

File: src/store/trails/mutations.js [hunk 1]
```diff
@@ -8,10 +8,9 @@ export const setFees = (state, config) => {
...
 
 export const resetBallots = (state) => {
-  console.log(`resetballots called`);
-  state.ballots.list.data = [];
-  state.ballots.list.loaded = false;
-  console.log(`ballots reset`);
+  // THere's no need to reset what's already fetched. Just use filters.
+  // state.ballots.list.data = [];
+  // state.ballots.list.loaded = false;
 };
```
Comment (donnyquixotic) [inline src/store/trails/mutations.js:14]: Is this _temporarily_ commented out? If not, we can remove unused method.

File: src/store/trails/mutations.js [hunk 2]
```diff
@@ -53,15 +52,17 @@ export const setBallot = (state, ballot) => {
...
   state.ballotVoters = voters;
 };
 
 export const resetTreasuries = (state) => {
-  state.treasuries.list.data = [];
-  state.treasuries.list.loaded = false;
+  // same as resetBallos
+  // state.treasuries.list.data = [];
+  // state.treasuries.list.loaded = false;
 };
```
Comment (donnyquixotic) [inline src/store/trails/mutations.js:67]: Same here, if we aren't using the method, we can just remove it.

**Review by donnyquixotic — CHANGES_REQUESTED** [review]

**Review by Kersten-TCD — APPROVED** [review]
Core functionality working, ready to launch.

Some smaller edge-case items found which will be added for next revision.

File: src/store/trails/mutations.js (hunk 1 above)
Comment (Kersten-TCD) [reply src/store/trails/mutations.js:14]: Please add to new issue as it does not seem to be a blocker for relaunch.

File: src/store/trails/mutations.js (hunk 2 above)
Comment (Kersten-TCD) [reply src/store/trails/mutations.js:67]: Please add to new issue as it does not seem to be a blocker for relaunch.

**Pull request merged**